repo/repo.go

```diff
@@ -97,6 +107,8 @@ func IngestRepo(ctx context.Context, bs cbor.IpldBlockstore, r io.Reader) (cid.C
 		}
 	}
 
+	bufrPool.Put(bufr)
```

Comment:
What happens if the function returns before this line? Do we need to defer putting the buffer back in the pool?

Reply:
yeah deferring the put probably makes sense, in general its not a huge deal if we miss putting something back in the pool in an error case, but this isnt inconvenient